include/cppoptlib/boundedproblem.h

```diff
@@ -52,6 +52,14 @@ class BoundedProblem : public Problem<Scalar_, CompileDim_> {
         setLowerBound(lb);
         setUpperBound(ub);
     }
+    
+    bool check_x(const TVector &x){
+        for (int i = 0; i < x.rows(); i++) {
+            if (x[i] < m_lowerBound[i] || x[i] > m_upperBound[i])
```

Comment:
Use the Eigen .any() / .all() functions instead of loops like this. It's clearer and should vectorize to be faster.